openfl/federated/task/runner_keras.py

```diff
@@ -298,10 +302,22 @@ def _get_weights_dict(obj, suffix=""):
             weights_dict (dict): The weight dictionary.
         """
         weights_dict = {}
-        weight_names = [weight.name for weight in obj.weights]
-        weight_values = obj.get_weights()
-        for name, value in zip(weight_names, weight_values):
-            weights_dict[name + suffix] = value
+        if isinstance(obj, ke.optimizers.Optimizer):
+            weight_names = [weight.name for weight in obj.variables]
+            weights_dict = {
+                weight_names[i] + suffix: weight.numpy()
+                for i, weight in enumerate(copy.deepcopy(obj.variables))
+            }
+        else:
+            weight_names = [
+                layer.name + "/" + weight.name for layer in obj.layers for weight in layer.weights
+            ]
```

Comment:
This code is repeated several times - could you extract it in a separate function?

Reply:
removed duplicate code.